app/scripts/modules/core/src/application/nav/NavCategory.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import { DataSourceNotifications } from 'core/entityTag/notifications/DataSourceNotifications';
+import { Icon } from '../../presentation';
+
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../application.model';
+import { IEntityTags } from 'core/domain';
+
+export interface INavCategoryProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+export const NavCategory: React.FC<INavCategoryProps> = ({ app, category, isActive }: INavCategoryProps) => {
+  const { alerts, badge, iconName, label } = category;
+  const badgeSourceData = badge ? app.getDataSource(badge).data.length : 0;
+
+  const [runningCount, setRunningCount] = React.useState<number>(badgeSourceData);
+  const [tags, setTags] = React.useState<IEntityTags[]>(alerts || []);
+
+  const runningCountSubscription = badge ? category.refresh$.subscribe(() => setRunningCount(badgeSourceData)) : null;
+  const tagsSubscription = category.refresh$.subscribe(() => setTags(alerts || []));
+  const [runningCountSub, setRunningCountSub] = React.useState(runningCountSubscription);
+  const [tagsSub, setTagsSub] = React.useState(tagsSubscription);
+
+  const configureSubscriptions = () => {
+    setRunningCountSub(runningCountSubscription);
+    setTagsSub(tagsSub);
+  };
+
+  const clearSubscriptions = () => {
+    runningCountSub && runningCountSub.unsubscribe();
+    tagsSub.unsubscribe();
+  };
+
+  React.useEffect(() => {
+    setRunningCount(badgeSourceData);
+    setTags(alerts || []);
+    clearSubscriptions();
+    configureSubscriptions();
+    return clearSubscriptions;
+  }, [alerts, badgeSourceData]);
```

Comment:
There's a lot of manual/imperative subscription and state management going on in this component, which I think we might be able to reduce. If I'm reading it right (please tell me if I'm not!) it appears the intent is roughly:
- If a `badge` is specified, use whatever the current data in the `badge`  data source is, and then subscribe to future updates
- Use whatever the current data in the `category` data source is to lookup `alerts`, and then subscribe to future updates on that `alerts` value

Based on that, I think you could likely make very good use of [the new `useDataSource` and `useObservable` hooks](https://github.com/spinnaker/deck/pull/8040) which handle the vast majority of the subscription + state management and let you mostly write stateless code in your actual component. I recommend playing with them and seeing if they work for you. e.g. for the badge one:

```tsx
// The default value is handled for you (it's whatever the current value of the data source is)
const { data: badgeData } = useDataSource(app.getDataSource(badge));
const runningCount = badgeData?.length || 0;
```

The only catch is I'm not 100% sure the hooks can handle passing an undefined/null value which you'd be doing here for `badge`, so might need to modify the hooks to check for nullish values and do nothing in those cases. If you run into that problem I'm happy to help get the hooks fixed.

Because `alerts` isn't part of the data source's status object, right now you'd probably still need a `useState` for that e.g.:

```tsx
const [alerts, setAlerts] = useState(category.alerts);
useObservable(category.status$, () => {
  setAlerts(category.alerts);
});
```

Reply:
Yes! Thank you, this is exactly what this needs. To be honest, migrating `DataSourceEntry` to this functional component already reduced a lot of the complexity so I am excited to clean it up a bit more.